rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2176,6 +2171,33 @@ private <T> T pollQueue() {
         }
     }
 
+    private void getKeyword(String variableName, StringBuilder keyword) {
+        String leftover = source.substring(cursor);
+        if (leftover.startsWith("final")) {
+            keyword.append("final");
+            cursor += 5;
+        } else if (leftover.startsWith("var")) {
+            keyword.append("var");
+            cursor += 3;
+        } else if (leftover.startsWith("def")) {
+            keyword.append("def");
+            cursor += 3;
+        } else if (leftover.startsWith(",")) {
+            keyword.append(",");
+            cursor++;
+        } else {
+            throw new IllegalStateException("Ran into unknown or unimplemented identifier");
```

Comment:
I'm not a fan of throwing exception, but if we do, it might be nice to include `leftover` here to reduce the guess work.

Reply:
https://github.com/openrewrite/rewrite/blob/bbdd436fd665a8c20a3878297de313ce9c17ffb6/rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java#L219-L222
Eventually the exception is caught and the 10 characters from where the cursor is are included in the checked exception.
Perhaps we should capture this in a test?

Reply:
I've seen that cursor be off quite a bit when there are parser issues; hence why I would include it in the thrown exception here when readily available. I don't think it's easy to trigger with a test, so let's not spend time on that.

Reply:
I did spend some time on it and it indeed isn't easy. Perhaps we could find a way to pass custom assertions with a `SourceSpec` but for now that's way out of scope and I'll just include a part of the remaining string as an indicator of where the exception occured!